File: Core Data/WWDC 2015/Whats_new_in_core_data/iPhoneCoreDataRecipes/Classes/WeightConverterViewController.m
/*
 Copyright (C) 2017 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 View controller to manage conversion of metric to imperial units of weight and vice versa.
  The controller uses two UIPicker objects to allow the user to select the weight in metric or imperial units.
 */

#import "WeightConverterViewController.h"

#import "MetricPickerController.h"
#import "ImperialPickerController.h"

@interface WeightConverterViewController ()

@property (nonatomic, strong) IBOutlet UIView *pickerViewContainer;

@property (nonatomic, strong) IBOutlet MetricPickerController *metricPickerController;
@property (nonatomic, strong) IBOutlet UIView *metricPickerViewContainer;

@property (nonatomic, strong) IBOutlet ImperialPickerController *imperialPickerController;
@property (nonatomic, strong) IBOutlet UIView *imperialPickerViewContainer;

@property (nonatomic, strong) IBOutlet UISegmentedControl *segmentedControl;

@property (nonatomic, assign) NSUInteger selectedUnit;

@end


#pragma mark -

@implementation WeightConverterViewController

#define METRIC_INDEX 0
#define IMPERIAL_INDEX 1

- (void)viewDidLoad {
    
	[super viewDidLoad];
		
	// Set the currently-selected unit for self and the segmented control.
	self.selectedUnit = METRIC_INDEX;
	self.segmentedControl.selectedSegmentIndex = self.selectedUnit;
	
	[self toggleUnit];
}

- (IBAction)toggleUnit {
	
	/*
	 When the user changes the selection in the segmented control, set the appropriate picker as the current subview of the picker container view (and remove the previous one).
	 */
	_selectedUnit = self.segmentedControl.selectedSegmentIndex;
	if (self.selectedUnit == IMPERIAL_INDEX) {
		[self.metricPickerViewContainer removeFromSuperview];
		[self.pickerViewContainer addSubview:self.imperialPickerViewContainer];
		[self.imperialPickerController updateLabel];
	} else {
		[self.imperialPickerViewContainer removeFromSuperview];
		[self.pickerViewContainer addSubview:self.metricPickerViewContainer];
		[self.metricPickerController updateLabel];
	}
}

@end

File: Core Data/Core Data Sample Codes/iPhoneCoreDataRecipes/Recipes_Swift_3_Updated/Recipes_Swift/Converter View Controllers/WeightConverterViewController.swift
/*
  WeightConverterViewController.swift
  Recipes_Swift

  Created by Seyed Samad Gholamzadeh on 2/28/18.
  Copyright © 2018 Seyed Samad Gholamzadeh. All rights reserved.

Abstract:
View controller to manage conversion of metric to imperial units of weight and vice versa.
The controller uses two UIPicker objects to allow the user to select the weight in metric or imperial units.
*/


import UIKit

class WeightConverterViewController: UIViewController {

	@IBOutlet weak var pickerViewContainer: UIView!
	@IBOutlet weak var metricPickerController: MetricPickerController!
	@IBOutlet weak var metricPickerViewContainer: UIView!
	@IBOutlet weak var imperialPickerController: ImperialPickerController!
	@IBOutlet weak var imperialPickerViewContainer: UIView!
	@IBOutlet weak var segmentedControl: UISegmentedControl!
	
	var selectedUnit: Int!

	enum Index: Int {
		case metric, imperial
	}

    override func viewDidLoad() {
        super.viewDidLoad()

		// Set the currently-selected unit for self and the segmented control.
		self.selectedUnit = Index.metric.rawValue
		self.segmentedControl.selectedSegmentIndex = self.selectedUnit
		self.toggleUnit()
    }
	
	@IBAction func toggleUnit() {
		
		/*
		When the user changes the selection in the segmented control, set the appropriate picker as the current subview of the picker container view (and remove the previous one).
		*/
		self.selectedUnit = self.segmentedControl.selectedSegmentIndex
		
		if self.selectedUnit == Index.imperial.rawValue {
			self.metricPickerViewContainer.removeFromSuperview()
			self.pickerViewContainer.addSubview(self.imperialPickerViewContainer)
			self.imperialPickerController.updateLabel()
		}
		else {
			self.imperialPickerViewContainer.removeFromSuperview()
			self.pickerViewContainer.addSubview(self.metricPickerViewContainer)
			self.metricPickerController.updateLabel()
		}

	}
	
}
